### PR TITLE
Fix pointer to compare the log level of Prelude and the current event

### PR DESCRIPTION
**Pull request by M3nace**
The pointer used for comparison was not updated.

|Related issue|
|---|
|#2552|

## Description

The pointer `lf` contains the `current_rule` object, which was used before, but not updated.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation

### CODE REVIEW

**Comment by DaveVG1** [comment]
Just out of curiosity, what's your use case for using the prelude library with Wazuh? @M3nace

**Comment by M3nace** [comment]
I'm a developer among the team of Prelude SIEM software.
Recently, we have some project using your solution, and since the connector with Prelude SIEM exists with OSSEC, we're maintaining this connector as well.

File: src/analysisd/analysisd.c [hunk 1]
```diff
@@ -1848,7 +1848,7 @@ void * w_writer_log_thread(__attribute__((unused)) void * args ){
     #ifdef PRELUDE_OUTPUT_ENABLED
                 /* Log to prelude */
                 if (Config.prelude) {
-                    if (Config.prelude_log_level <= currently_rule->level) {
+                    if (Config.prelude_log_level <= lf->currently_rule->level) {
```
Comment (DaveVG1) [inline src/analysisd/analysisd.c]: This won't compile. The `Eventinfo `struct doesn't have a member called `currently_rule`. It's called `generated_rule` instead. This should be:

```
if (Config.prelude_log_level <= lf->generated_rule->level) {
```

Comment (M3nace) [reply src/analysisd/analysisd.c]: My bad, fixing that rn.

**Comment by DaveVG1** [comment]
> I'm a developer among the team of Prelude SIEM software.
> Recently, we have some project using your solution, and since the connector with Prelude SIEM exists with OSSEC, we're maintaining this connector as well.

Glad to hear that! I hope the changed requested works for you. Thanks for your interest!

**Comment by M3nace** [comment]
Thanks for your review :)

**Comment by M3nace** [comment]
Hello,
May I have an update on this ? The changes requested have been done.
Thank you very much

File: src/analysisd/analysisd.c [hunk 2]
```diff
@@ -1848,7 +1848,7 @@ void * w_writer_log_thread(__attribute__((unused)) void * args ){
     #ifdef PRELUDE_OUTPUT_ENABLED
                 /* Log to prelude */
                 if (Config.prelude) {
-                    if (Config.prelude_log_level <= currently_rule->level) {
+                    if (Config.prelude_log_level <= lf->generated_rule->level) {
```
Comment (DaveVG1) [inline src/analysisd/analysisd.c]: We should also check if `lf->generated_rule` is not `NULL` in order to avoid a possible segmentation fault.

**Comment by DaveVG1** [comment]
Hi @M3nace,

This PR is set to be merged for `v3.11`.  However, I added another minor revision for the code. Sorry for the inconvenience and thank you so much again for your contribution.

Best regards,
David

File: src/analysisd/analysisd.c (hunk 2 above)
Comment (M3nace) [reply src/analysisd/analysisd.c]: Of course!

**Comment by M3nace** [comment]
Hi @DaveVG1,

Great news then ! I'm glad to hear that.
No problem, feel free the ask anything around Prelude.

Best regards,
Camille

**Comment by M3nace** [comment]
Pinging @DaveVG1, @bah07 
Any news about this PR ? 😃

**Pull request closed without merging**